core/src/main/java/org/commonjava/indy/core/expire/ScheduleManager.java

```diff
@@ -367,6 +369,57 @@ else if ( store instanceof Group )
         }
     }
 
+    public synchronized void rescheduleDisableTimeout( final StoreKey key )
+            throws IndySchedulerException
+    {
+        if ( !schedulerConfig.isEnabled() )
+        {
+            logger.debug( "Scheduler disabled." );
+            return;
+        }
+
+        ArtifactStore store = null;
+        try
+        {
+            store = dataManager.getArtifactStore( key );
+        }
+        catch ( final IndyDataException e )
+        {
+            logger.error( String.format( "Failed to retrieve store for: %s. Reason: %s", key, e.getMessage() ), e );
+        }
+
+        if ( store == null )
+        {
+            return;
+        }
+
+        int timeout = store.getDisableTimeout();
+        if ( timeout == TIMEOUT_USE_DEFAULT )
+        {
+            // case TIMEOUT_USE_DEFAULT: will use default timeout configuration
+            timeout = config.getStoreDisableTimeoutSeconds();
+        }
+
+        // No need to cancel as the job will be cancelled immediately after the re-enable in StoreEnablementManager
+//        final Set<ScheduleKey> canceled =
+//                cancelAllBefore( new StoreKeyMatcher( store.getKey(), DISABLE_TIMEOUT ),
+//                                 timeout );
+//        logger.info( "Cancel disable timeout for stores:{}", canceled );
+
+        if ( timeout > TIMEOUT_USE_DEFAULT && store.isDisabled() )
+        {
+            final StoreKey sk = store.getKey();
+            logger.debug( "Set/Reschedule disable timeout for store:{}", sk );
+            scheduleForStore( sk, DISABLE_TIMEOUT, DISABLE_TIMEOUT, sk, timeout );
+        }
+        else if ( timeout <= TIMEOUT_NEVER_DISABLE )
+        {
+            logger.debug( "Disable-timeout set to {}, will never disable the repo", timeout );
+            store.setDisabled( false );
```

Comment:
This line worries me a bit. It's maybe the only place where the ScheduleManager directly changes something about the store definition, and it doesn't try to save its changes. Knowing that this comes from a pre-store event (pre-disable) which is observed in another class is what allows this to be okay in a technical sense, but if anything else ever called this method, this re-enabling effect might be lost.

Is there a way we can send a clear signal to the calling code to say that the repository should not be disabled? I'm thinking of an enum return value, though that's sort of heavy when there are just two possible results. The problem with using a boolean, though, is that the two results aren't exactly directly opposed:

* Yes, timeout scheduled
* No, please don't disable this store

The "please don't disable this store" part is what would be confusing if we used a boolean, since it's not obvious from a false result.

Maybe an enum would be better, since we could also return SCHEDULER_DISABLED as another value...though that implies we should be doing that in other cases where we schedule timeouts, too.